### PR TITLE
(refactor): pass fn as first argument to nested_map to follow convention and remove redundant logic from nested functions

### PR DESCRIPTION
**Pull request by mattbarrett98**
Some cleaning up of nested functions, removed functionality relating to `max_depth` and `extra_nest_types` which just seem to be there for the sake of it, and have no uses in any repos.

Also changing the order of arguments in `nested_map` to pass the function to apply in first, which is consistent with python's `map` function, and basically any other equivalent function you can find (for example https://www.tensorflow.org/api_docs/python/tf/nest/map_structure , https://jax.readthedocs.io/en/latest/_autosummary/jax.tree_util.tree_map.html).

Let me know your thoughts @vedpatwardhan

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/ivy/gradients.py [hunk 1]
```diff
@@ -279,16 +277,16 @@ def _process_func_ret_and_grads(func_ret, grads, retain_grads):
 def _variable(x):
     x = ivy.to_native(x, nested=True)
     ret = ivy.nested_map(
-        x, current_backend(x).variable, include_derived=True, shallow=False
+        current_backend(x).variable, x, include_derived=True, shallow=False
     )
     return ivy.nested_map(ret, ivy.to_ivy, include_derived=True)
```
Comment (vedpatwardhan) [inline ivy/functional/ivy/gradients.py:282]: could you please change the function call here as well? Rest seems good to merge, thanks @mattbarrett98 😄

**Review by vedpatwardhan — APPROVED** [review]

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Feel free to merge, thanks @mattbarrett98 😄
(I went through most of the failing tests in the CI and they don't seem to be related to these changes)
